basex/src/main/java/com/formulasearchengine/mathosphere/basex/TexQueryGenerator.java

```diff
@@ -11,81 +18,102 @@
-		response = httpClient.execute( httppost );
-
-		HttpEntity entity = response.getEntity();
-
-		InputStream instream = entity.getContent();
-		ob = new ObjectMapper().readValue( instream, Map.class );
-
-		if ( ob.get( "status_code" ) == null || ob.get( "result" ) == null ) {
-			throw new IOException( "Unable to process MathML conversion server response to Tex request.");
-		}
-
-		if ( Integer.parseInt(  ob.get( "status_code" ).toString() ) > 1 ) {
-			throw new IOException( "Tex request to MathML conversion server produced failed response.",
-					new IOException( ob.get( "result" ).toString()));
-		}
-
-		return ob.get( "result" ).toString();
-	}
+
+    private List<NameValuePair> params = getDefaultParams();
+    private LateXMLConfig lateXMLConfig = new LateXMLConfig()
+            .setUrl("http://drmf-latexml.wmflabs.org")
+            .setParams(updateParamFormat(this.getParams()));
+    private LaTeXMLServiceResponse serviceResponse;
+
+
+    List<NameValuePair> getParams() {
+        return params;
+    }
+
+    @NotNull
+    private ArrayList<NameValuePair> getDefaultParams() {
+        ArrayList<NameValuePair> pDefault = new ArrayList<>();
+        pDefault.add(new BasicNameValuePair("format", "xhtml"));
+        pDefault.add(new BasicNameValuePair("whatsin", "math"));
+        pDefault.add(new BasicNameValuePair("whatsout", "math"));
+        pDefault.add(new BasicNameValuePair("cmml", ""));
+        pDefault.add(new BasicNameValuePair("nodefaultresources", ""));
+        pDefault.add(new BasicNameValuePair("preload", "LaTeX.pool"));
+        pDefault.add(new BasicNameValuePair("preload", "article.cls"));
+        pDefault.add(new BasicNameValuePair("preload", "amsmath.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amsthm.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amstext.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amssymb.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "eucal.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "[dvipsnames]xcolor.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "url.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "hyperref.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "mws.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "texvc"));
+        return pDefault;
+    }
+
+    void setParams(List<NameValuePair> params) {
+        this.params = params;
+        lateXMLConfig.setParams(updateParamFormat(params));
+    }
+
+    private Map<String, Object> updateParamFormat(List<NameValuePair> params) {
+        HashMap<String, Object> map = new HashMap<>();
+        for (NameValuePair pair : params) {
+            if (map.containsKey(pair.getName())) {
+                Object oSetting = map.get(pair.getName());
+                if (oSetting instanceof List) {
+                    ((List<String>) oSetting).add(pair.getValue());
+                } else {
```

Comment:
I needed some time that I did not add a
```java
else if (o instanceof Map)
```
case here. I finally concluded that there are no second order map parameters in LaTeXML. I hope that's correct. I looked this up at several places:
* https://github.com/wikimedia/mediawiki-extensions-Math/blame/2bbb86b868b9b5412d40df87af832d95aa4d8d0c/MathLaTeXML.php#L25-L44
* https://github.com/dginev/LaTeXML-Plugin-ltxpsgi/blob/master/script/ltxpsgi#L90-L122
* https://github.com/brucemiller/LaTeXML/blob/4d39f5b64a3a52c02c8a3d5b75ab0fefc2e181ca/lib/LaTeXML/Common/Config.pm#L35-L243